lib/management/index.js

```diff
@@ -164,28 +164,42 @@ function _saveServiceState(newState, cb) {
                                 JSON.stringify(newState), {}, logger, cb);
 }
 
+function _applyWorkflowUpdate(conf, currentState, params, workflow, cb) {
```

Comment:
Sorry if I'm being nitpicky here, I think this helper is confusing, since it has almost the same name than another function that it calls but does more things. Why not simply call `getWorkflowUpdates` from the main `async.each` loop before calling `applyWorkflowUpdates`?

Reply:
I agree with that—will update.

Reply:
Ok, I've renamed the method called from `_refreshServiceState` to `applyServiceState`. `applyServiceState` will abstract the process of getting the workflow and applying them. I felt it helps with readability since the nesting was getting deep, having added the `async.each`.